src/utils/backend/vaccine.ts

```diff
@@ -0,0 +1,83 @@
+import { BackendDataReturn, DatabaseClient } from "@utils/types/common";
+import { VaccineRecord } from "@utils/types/vaccine";
+
+export async function getVaccineRecordbyPersonId(
+  supabase: DatabaseClient,
+  personId: number
+): Promise<VaccineRecord[]> {
+  const { data, error } = await supabase
+    .from("vaccine_records")
+    .select("*")
+    .eq("person", personId)
+    .order("vaccination_date", { ascending: true });
+
+  if (error) {
+    throw error;
+  }
+
+  return (
+    data!.map((vaccineRecord, index) => ({
+      id: vaccineRecord.id,
+      doseNo: index + 1,
+      vaccineName: vaccineRecord.vaccine_name,
+      vaccineDate: new Date(vaccineRecord.vaccination_date),
+      lotNo: vaccineRecord.lot_no,
+      administeredBy: vaccineRecord.administering_center,
+    })) ?? []
+  );
+}
```

Comment:
Now that you brought it up, holy bananas 🍌🍌 is this a better way to do this. Death to `database.ts`!